packages/gatsby-plugin-sitemap/README.md

```diff
@@ -21,89 +21,190 @@ plugins: [`gatsby-plugin-sitemap`]
+- [`serialize`](#serialize) (function) Takes the output of the data query and lets you return an array of sitemap entries.
 
-We _ALWAYS_ exclude the following pages: `/dev-404-page`,`/404` &`/offline-plugin-app-shell-fallback`, this cannot be changed.
+The following pages are _ALWAYS_ excluded: `/dev-404-page`,`/404` &`/offline-plugin-app-shell-fallback`, this cannot be changed even by customizing the [`filterPages`](#filterPages) function.
 
-Example:
+## Example:
 
 ```javascript
+const siteUrl = process.env.URL || `https://fallback.net`
+
 // In your gatsby-config.js
-siteMetadata: {
-  siteUrl: `https://www.example.com`,
-},
-plugins: [
-  {
-    resolve: `gatsby-plugin-sitemap`,
-    options: {
-      output: `/some-other-sitemap.xml`,
-      // Exclude specific pages or groups of pages using glob parameters
-      // See: https://github.com/isaacs/minimatch
-      // The example below will exclude the single `path/to/page` and all routes beginning with `category`
-      exclude: [`/category/*`, `/path/to/page`],
-      query: `
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-plugin-sitemap",
+      options: {
+        query: `
         {
-          wp {
-            generalSettings {
-              siteUrl
-            }
-          }
-
           allSitePage {
             nodes {
               path
             }
           }
-      }`,
-      resolveSiteUrl: ({site, allSitePage}) => {
-        //Alternatively, you may also pass in an environment variable (or any location) at the beginning of your `gatsby-config.js`.
-        return site.wp.generalSettings.siteUrl
-      },
-      serialize: ({ site, allSitePage }) =>
-        allSitePage.nodes.map(node => {
+          allWpContentNode(filter: {nodeType: {in: ["Post", "Page"]}}) {
+            nodes {
+              ... on WpPost {
+                uri
+                modifiedGmt
+              }
+              ... on WpPage {
+                uri
+                modifiedGmt
+              }
+            }
+          }
+        }
+      `,
+        resolveSiteUrl: () => siteUrl,
+        resolvePages: ({
+          allSitePage: { nodes: allPages },
+          allWpContentNode: { nodes: allWpNodes },
+        }) => {
+          const wpNodeMap = allWpNodes.reduce((acc, node) => {
+            const { uri } = node
+            acc[uri] = node
+
+            return acc
+          }, {})
+
+          return allPages.map(page => {
+            return { ...page, ...wpNodeMap[page.path] }
+          })
+        },
+        serialize: ({ path, modifiedGmt }) => {
           return {
-            url: `${site.wp.generalSettings.siteUrl}${node.path}`,
-            changefreq: `daily`,
-            priority: 0.7,
+            url: path,
+            lastmod: modifiedGmt,
           }
-        })
-    }
-  }
-]
+        },
+      },
+    },
+  ],
+}
+```
+
+## API Reference
+
+<a id=resolveSiteUrl></a>
```

Comment:
because I wasn't about to try and figure out and remember how `## serialize ⇒ <code>object</code>` converted to an id. It was easier to create anchors that just had the function name to easily link to. If there is an A11y or other concern I'm unaware of we can find a better way or just do it the hard way.